### ISSUE TITLE
Misuses of cryptographic APIs

### ISSUE THREAD

**Issue by mhp0rtal** [joyent/java-http-signature#55]
Hi
The following lines have cryptographic API misuses. <br> File name => common/src/main/java/com/joyent/http/signature/KeyFingerprinter.java: Line number => 64: API name => MessageDigest:<BR> File name => common/src/main/java/com/joyent/http/signature/crypto/JceDigest.java: Line number => 35: API name => MessageDigest:Operation on object of type java.security.MessageDigest object not completed. Expected call to digest, update<BR> File name => common/src/main/java/com/joyent/http/signature/Signer.java: Line number => 122: API name => Signature:Operation on object of type java.security.Signature object not completed. Expected call to initSign, initVerify<BR> File name => common/src/main/java/com/joyent/http/signature/Signer.java: Line number => 116: API name => Signature:Operation on object of type java.security.Signature object not completed. Expected call to initSign, initVerify<BR> File name => common/src/main/java/com/joyent/http/signature/KeyFingerprinter.java: Line number => 55: API name => MessageDigest:First parameter (with value "MD5") should be any of {SHA-256, SHA-384, SHA-512}<BR>

**Comment by jclulow** [comment]
It's a bit hard to follow this.  Can you reformat it to make it clearer what problems you're seeing in which files, and what the fix ought to be?
